### PR TITLE
Johnshew Dev

### LINKED ISSUE #9: Clean up and unit test app web server

### PR DESCRIPTION
**Pull request by johnshew**
Convert to vueify
Edit capability for tasks from web app
Edit capability for conversations from web app
Add support for hostname on authentication redirects
Updated LUIS app with additional intents
Add mocks for App (partially complete but tests pass)
Experiment with parallel conversation plans - compiles but not called
Experiment with TurnGenerator - currently commented out
Bug fixes and error cleanup
Fixes #9, #10, #28, #31, #33, #40

### CODE REVIEW

**Comment by johnshew** [comment]
James ignore for now.  I'll pull across the changes that are in johnshew/typescript-jest-modules to remove the dependency on .env that got reintroduced with the httpServer test and will then resubmit.

**Comment by Jatrott** [comment]
No problem, sorry I missed your text, I left my phone on my desk and by the time I saw the messages it seemed a little late to disturb you. (Given the timestamps I'm assuming this thread is what it was about)


- James


Sent from Outlook<http://aka.ms/weboutlook>

________________________________

**Comment by johnshew** [comment]
@Jatrott ready to go now

File: src/httpServer.ts [hunk 1]
```diff
@@ -170,26 +176,80 @@ function configureServer(httpServer: restify.Server) {
 
     // APIs - no html - just json response
 
-    httpServer.get('/api/v1.0/tasks', async (req, res, next) => {
-        await graphGet(req, res, next, `https://graph.microsoft.com/beta/me/outlook/tasks?${app.graph.filterNotCompletedAndNagMeCategory}&${app.graph.queryExpandNagExtensions}`);
+
+    httpServer.get('/api/v1.0/me', async (req, res, next) => {
+        await graphGet(req, res, next, "https://graph.microsoft.com/v1.0/me");
```
Comment (Jatrott) [inline src/httpServer.ts]: feels like some of these endpoints should be configuration based, not inline in code.

File: src/httpServer.ts [hunk 2]
```diff
@@ -170,26 +176,80 @@ function configureServer(httpServer: restify.Server) {
...
+            res.end();
+            return next();
+        } catch (err) { error = error }
+        res.status(400);
+        res.json({ error });
+        res.end();
+        return next();
+    })
 
     /// Interactive Tests
```
Comment (Jatrott) [inline src/httpServer.ts]: what are interactive tests? is their test specific code in the httpserver?

File: src/simpleAuth.ts [hunk 3]
```diff
@@ -43,28 +43,30 @@ export class AuthContext {
...
 
-    private userAuthKeyToTokensMap = new Map<string, AuthContext>(); // UserAuthKey to AuthTokens
+    userAuthKeyToTokensMap = new Map<string, AuthContext>(); // UserAuthKey to AuthTokens
 
-    constructor(private appId: string, private appPassword: string, private defaultRedirectUri: string, private scopes: string[] = []) {
+    constructor(private appId: string, private appPassword: string, private redirectUrl: string = "error", private scopes: string[] = []) {
         super();
     }
 
     // Clients of the authManager interact with it using an opaque AuthKey (string) or OID (string).
```
Comment (Jatrott) [inline src/simpleAuth.ts]: this comment block might be better interspersed with code, so its easier to conceptually map for a new reader.

File: src/httpServer.ts (hunk 1 above)
Comment (johnshew) [reply src/httpServer.ts]: Agree #42

File: src/httpServer.ts (hunk 2 above)
Comment (johnshew) [reply src/httpServer.ts]: To be removed

File: src/simpleAuth.ts (hunk 3 above)
Comment (johnshew) [reply src/simpleAuth.ts]: Makes sense

**Pull request merged**